Sources/AsyncQueue/ActorQueue.swift

```diff
@@ -50,8 +50,9 @@
 /// }
 /// ```
 ///
+/// - Warning: The `ActorQueue`'s conformance to `@unchecked Sendable` is safe if and only if `adoptExecutionContext(of:)` is called only from the adopted actor's `init` method.
 /// - Precondition: The lifecycle of an `ActorQueue` must not exceed that of the adopted actor.
-public final class ActorQueue<ActorType: Actor> {
+public final class ActorQueue<ActorType: Actor>: @unchecked Sendable {
```

Comment:
If we don't make `ActorQueue` be `Sendable`, then we get the following error in consuming code whenever the `ActorQueue` instance is accessed from a `nonisolated` method:
```
Non-sendable type 'ActorQueue<ActorType>' in asynchronous access to actor-isolated property 'actorQueue' cannot cross actor boundary
```

This error makes sense: we're accessing a non-sendable `let` constant outside of the isolated context. I'm looking forward to Swift 6 when I don't need to manually turn on this checking to see these kinds of issues in my packages.

Reply:
We also can't make this type `Sendable` because `weakExecutionContext` is mutable and therefore the type _is not Sendable_. But given the API contract of:
1. `adoptExecutionContext(of:)` must be called from `init`
2. `adoptExecutionContext(of:)` must not be called more than once
3. An `ActorQueue` should be a `private let` constant on the adopted actor instance

The `ActorQueue` is effectively Sendable, and therefore we can mark it as such.